hooks/build

```
@@ -0,0 +1,449 @@
+    done
+
+    echo
+    echo "This build uses Docker cache"
+    echo "Cache directive used: ${DOCKER_CACHE_DIRECTIVE}"
+    echo
+elif [[ "${AIRFLOW_CONTAINER_USE_NO_CACHE}" == "true" ]]; then
+    DOCKER_CACHE_DIRECTIVE="--no-cache"
+    echo
+    echo "Skip cache for builds. Everything will be rebuilt from scratch."
+    echo
+    echo "Cache directive used: ${DOCKER_CACHE_DIRECTIVE}"
+    echo
+else
+    DOCKER_CACHE_DIRECTIVE="--cache-from ${AIRFLOW_CI_IMAGE} --cache-from ${AIRFLOW_IMAGE}"
+    echo
+    echo "Use default cache from locally built images."
+    echo
+    echo "Cache directive used: ${DOCKER_CACHE_DIRECTIVE}"
+    echo
+fi
+
+STAT_BIN=stat
+if [[ "${OSTYPE}" == "darwin"* ]]; then
+    STAT_BIN=gstat
+fi
+# Build id identifying the build uniquely
+BUILD_ID=${BUILD_ID:="local"}
+# Branch name for triggered builds
+BRANCH_NAME=${BRANCH_NAME:="master"}
+
+# directory where "deployment" artifacts should be placed
+DEPLOY_DIR=${AIRFLOW_ROOT}/dist/${BRANCH_NAME}/$(date +%Y-%m-%d)/${BUILD_ID}/${PYTHON_VERSION}
+
+mkdir -p "${DEPLOY_DIR}"
+
+FIXED_PERMISSIONS_FILE=${DEPLOY_DIR}/fixed-permissions.txt
+ALL_FILES_FILE=${DEPLOY_DIR}/all-files.txt
+ALL_FILES_IN_CONTEXT_FILE=${DEPLOY_DIR}/all-files-in-context.txt
+
+mkdir -pv ${DEPLOY_DIR}
+
+if  [[ "${AIRFLOW_FIX_PERMISSIONS}" == "all" || "${AIRFLOW_FIX_PERMISSIONS}" == "setup" ]]; then
+    start_step "Fixing permissions for CI builds for ${AIRFLOW_FIX_PERMISSIONS} files"
+    if [[ "${AIRFLOW_FIX_PERMISSIONS}" == "all" ]]; then
+        # Get all files in the context - by building a small alpine based image
+        # then COPY all files (.) from the context and listing the files via find method
+        if [[ "$(docker images -q airflow-context:latest 2> /dev/null)" == "" ]]; then
+            docker build -t airflow-context:latest .  -f Dockerfile-context
+        fi
+        ALL_FILES_TO_FIX=$(docker run airflow-context:latest /bin/sh -c "(cd /context && find .)")
+    elif [[ "${AIRFLOW_FIX_PERMISSIONS}" == "setup" ]]; then
+        ALL_FILES_TO_FIX="\
+            ${AIRFLOW_ROOT}/setup.py \
+            ${AIRFLOW_ROOT}/setup.cfg \
+            ${AIRFLOW_ROOT}/airflow/version.py \
+            ${AIRFLOW_ROOT}/airflow/__init__.py \
+            ${AIRFLOW_ROOT}/airflow/bin/airflow"
+    fi
+    STAT_BIN=stat
+    if [[ "${OSTYPE}" == "darwin"* ]]; then
+        STAT_BIN=gstat
+    fi
+
+    for FILE in ${ALL_FILES_TO_FIX}; do
+        ACCESS_RIGHTS=$("${STAT_BIN}" -c "%A" "${FILE}" || echo "--------")
+        # check if the file is group/other writeable
+        if [[ "${ACCESS_RIGHTS:5:1}" != "-" || "${ACCESS_RIGHTS:8:1}" != "-" ]]; then
+            "${STAT_BIN}" --printf "%a %A %F \t%s \t->    " "${FILE}"
+            chmod og-w "${FILE}"
+            "${STAT_BIN}" --printf "%a %A %F \t%s \t%n\n" "${FILE}"
+        fi
+    done
+
+    echo "Group/other write access removed for all $(echo ${ALL_FILES_TO_FIX} | wc -w) files in context"
```

```suggestion
    echo "Group/other write access removed for all $(echo "${ALL_FILES_TO_FIX}" | wc -w) files in context"
```
Double quote to prevent globbing and word splitting